src/main/host/syscall/handler/sysinfo.rs

```diff
@@ -0,0 +1,53 @@
+use crate::core::support::emulated_time;
+use crate::core::worker::Worker;
+use crate::host::context::ThreadContext;
+use crate::host::syscall::handler::SyscallHandler;
+use crate::host::syscall_types::{PluginPtr, SysCallArgs, SyscallResult, TypedPluginPtr};
+use crate::utility::pod;
+
+use nix::errno::Errno;
+use syscall_logger::log_syscall;
+
+impl SyscallHandler {
+    // Using libc::c_void here because the libc::sysinfo struct is a return value only.
+    #[log_syscall(/* rv */ libc::c_int, /* info */ *const libc::sysinfo)]
+    pub fn sysinfo(&self, ctx: &mut ThreadContext, args: &SysCallArgs) -> SyscallResult {
+        // Pointer to the plugin memory where we write the result.
+        let info_ptr = TypedPluginPtr::new::<libc::sysinfo>(PluginPtr::from(args.get(0)), 1);
+        if info_ptr.is_null() {
+            return Err(Errno::EFAULT.into());
+        }
+
+        // Seconds are needed for uptime.
+        let time = Worker::current_time().unwrap_or(emulated_time::SIMULATION_START);
```

Comment:
I think it would be better to `unwrap()` since if the worker is not initialized, then something bad has happened and we probably want to catch it early rather than continue and return the wrong value.

Reply:
Maybe we don't want `Worker::current_time()` to return `Option` then, if we're going to always unwrap it anyway?

Reply:
I'm not sure what the original reason for returning an `Option` was, but since the `Worker` is a global that can be accessed from anywhere within Shadow, I think it makes sense to keep it an `Option` since there might be cases where we don't know if the global worker is set or not and we want to handle both cases. For example the logger gets the current time, and there might not be a worker active at the time. In this case, it's specifically an error if the worker isn't set during a syscall handler.